@types/index.d.ts

```diff
@@ -142,6 +142,8 @@ declare class Request extends Body {
 	clone(): Request;
 }
 
+type ResponseType = "basic" | "cors" | "default" | "error" | "opaque" | "opaqueredirect"
```

Comment:
Why do we want to include type values that we will never actually return?

Reply:
I've made the same comment above.
Let's settle this:

- Use ResponseType from TypeScript lib.dom.d.ts
  Votes: none 
- Create our own with just `type ResponseType = "default" | "error"`
  Votes: 
  - my preference
  - @tinovyatkin 
- Create our own with everything: `type ResponseType = "basic" | "cors" | "default" | "error" | "opaque" | "opaqueredirect"`
  Votes:
  - @tekwiz 
  
What do we do?

Reply:
@tinovyatkin My argument is this: just because we don't use them doesn't mean they're disallowed.  In addition to maintaining parity with the spec, it should remain possible for someone to extend node-fetch to include filtered responses, and limiting the ResponseType seems unnecessarily restrictive.  That said, it is entirely possible that my limited understanding of TypeScript is leading me to the wrong conclusion about the complexity of adding to the type.